File: DemoMMOUnity/Assets/Game/Net/Script/Net/CsProtocol/Protocol/ServerConfig/ServerConfigRequest.cs
using System.Collections.Generic;
using Newtonsoft.Json;
using ZJYFrameWork.Net.Core;

namespace ZJYFrameWork.Net.CsProtocol.Buffer.Protocol.ServerConfig
{
    /// <summary>
    /// 请求服务器上面配置表
    /// </summary>
    public class ServerConfigRequest : Model, IPacket
    {
        public static ServerConfigRequest ValueOf()
        {
            var packet = new ServerConfigRequest();

            return packet;
        }


        public short ProtocolId()
        {
            return 1009;
        }
    }

    public class ServerConfigRequestRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 1009;
        }

        public void Write(ByteBuffer buffer, IPacket packet)
        {
            ServerConfigRequest message = (ServerConfigRequest)packet;
            var _message = new ServerMessageWrite(message.ProtocolId(), message);
            var json = JsonConvert.SerializeObject(_message);
            buffer.WriteString(json);
        }

        public IPacket Read(ByteBuffer buffer)
        {
            return null;
        }
    }
}

File: DemoMMOUnity/Lua/application/app/net/webscoket/luaProtocol/serverConfig/serverConfigRequest.lua
---@class ServerConfigRequest
local ServerConfigRequest = class("ServerConfigRequest")
function ServerConfigRequest:ctor()
    ---@type string
    self.panel = string.empty
end

---@param panel string
function ServerConfigRequest:new(panel)
    self.panel = panel
    return self
end

function ServerConfigRequest:protocolId()
    return 1009
end

function ServerConfigRequest:write()

    local message = {
        protocolId = self:protocolId(),
        packet =
        { panel = self.panel }
    }
    local jsonStr = JSON.encode(message)
    return jsonStr
end

function ServerConfigRequest:read(buffer)
    local jsonString = buffer:readString()
    ---字节读取器中存放字符
    local data = JSON.decode(jsonString)
    return ServerConfigRequest:new(data.packet.panel)
end

return ServerConfigRequest
